src/App.js

```diff
@@ -1,7 +1,151 @@
-import "./App.css";
+import "./App.css"
+import { Component } from 'react'
+import productData from "./data/productData"
+import formatPrice from "./helpers/formatPrice"
 
-const App = () => {
-  return <h1>Hello, world!</h1>;
-};
+class App extends Component {
+  constructor(){
+    super()
+    this.state = {
+      products: productData,
+      cart:{subtotal: 0, tax: 0, total: 0},
+      checkout:{firstName:"", lastName:"", email:"", creditCard:"", zipCode:""},
+      cartItems:[]
+    }
+  }
 
-export default App;
+  handleFirstName=(event)=>{
+    this.setState({
+        checkout:{...this.state.checkout,firstName: event.target.value}
+    })
+  }
+
+  handleLastName=(event)=>{
+    this.setState({
+        checkout:{...this.state.checkout,lastName: event.target.value}
+    })
+  }
+
+  handleEmail=(event)=>{
+    this.setState({
+        checkout:{...this.state.checkout,email: event.target.value}
+    })
+  }
+
+  handleCard=(event)=>{
+    this.setState({
+        checkout:{...this.state.checkout,creditCard: event.target.value}
+    })
+  }
+
+  handleZipCode=(event)=>{
+    this.setState({
+        checkout:{...this.state.checkout,zipCode: event.target.value}
+    })
+  }
```

Comment:
```suggestion
handleInput = (event) => {
    this.setState({
      checkout: {...this.state.checkout, [event.target.name]: event.target.value}
    })
}    
```
This is how you can condense those five input event handlers into a single one. Make sure that the "name" on each input matches exactly with their name in state, or else this won't work.